Add documentation for user-configurable overrides module (#3374)
grafana/tempo: docs/sources/tempo/api_docs/_index.md

Meant to be used in a Query Visualization UI like Grafana to test that the Tempo data source is working.
{{% /admonition %}}

### Overrides API

> The user-configurable overrides API is documented in detail at [user-configurable overrides]{{< relref "../operations/user-configurable-overrides#api" >}}.

Reviewer: ```suggestion
For more information about user-configurable overrides API, refer to the [user-configurable overrides]{{< relref "../operations/user-configurable-overrides#api" >}} documentation.
```
